openid-4-verifiable-presentations-1_0.md

```diff
@@ -1549,6 +1549,90 @@ Note: The reason hashes of the user claims are included in the `issuerAuth` item
 
 The example in this section is also applicable to the electronic identification Verifiable Credentials expressed using data models defined in ISO/IEC TR 23220-2.
 
+## IETF SD-JWT VC
+
+This section defines how credentials complying with [@!I-D.ietf-oauth-sd-jwt-vc] can be presented to the Verifier using this specification.
+
+### Format Identifier
+
+The Credential format identifier is `vc+sd-jwt`.
+
+#### Example Credential
+
+The following is a non-normative example of the payload of an IETF SD-JWT VC that will be used throughout this section:
+
+<{{examples/credentials/sd_jwt_vc.json}}
+
+The following are disclosures belonging to the claims from the example above.
+
+__Claim `given_name`__:
+
+ * SHA-256 Hash: `jsu9yVulwQQlhFlM_3JlzMaSFzglhQG0DpfayQwLUK4`
+ * Disclosure:\
+`WyIyR0xDNDJzS1F2ZUNmR2ZyeU5STjl3IiwgImdpdmVuX25hbWUiLCAiSm9o`\
+`biJd`
+ * Contents:
+`["2GLC42sKQveCfGfryNRN9w", "given_name", "John"]`
+
+
+__Claim `family_name`__:
+
+ * SHA-256 Hash: `TGf4oLbgwd5JQaHyKVQZU9UdGE0w5rtDsrZzfUaomLo`
+ * Disclosure:\
+`WyJlbHVWNU9nM2dTTklJOEVZbnN4QV9BIiwgImZhbWlseV9uYW1lIiwgIkRv`\
+`ZSJd`
+ * Contents:
+`["eluV5Og3gSNII8EYnsxA_A", "family_name", "Doe"]`
+
+
+__Claim `birthdate`__:
+
+ * SHA-256 Hash: `tiTngp9_jhC389UP8_k67MXqoSfiHq3iK6o9un4we_Y`
+ * Disclosure:\
+`WyI2SWo3dE0tYTVpVlBHYm9TNXRtdlZBIiwgImJpcnRoZGF0ZSIsICIxOTQw`\
+`LTAxLTAxIl0`
+ * Contents:
+`["6Ij7tM-a5iVPGboS5tmvVA", "birthdate", "1940-01-01"]`
+
+### Verifier Metadata
+
+The Verifier SHOULD add a `vp_formats` element to its metadata (e.g. in the `client_metadata` authorization request parameter) to let the wallet know what protection algorithms it supports in conjunction with SD-JWT VCs. The format element MUST have the key `vc+sd-jwt`, the value is an object consisting of the following elements:
+
+* `sd-jwt_alg_values`: OPTIONAL. A JSON array containing identifiers of cryptographic algorithms the verifier supports for protection of a SD-JWT. If present, the `alg` JOSE header (as defined in [@!RFC7515]) of the presented SD-JWT MUST match one of the array values.
+* `kb-jwt_alg_values`: OPTIONAL. A JSON array containing identifiers of cryptographic algorithms the verifier supports for protection of a KB-JWT. If present, the `alg` JOSE header (as defined in [@!RFC7515]) of the presented KB-JWT MUST match one of the array values.
+
+The following is a non-normative example of `client_metadata` request parameter value in a request to present a SD-JWT VC.
+
+<{{examples/client_metadata/sd_jwt_vc_verifier_metadata.json}}
+
+### Presentation Request
+
+The following is a non-normative example of an Authorization Request:
+
+<{{examples/request/request.txt}}
```

Comment:
The data (e.g., nonce, client_id) in this request doesn't align with the data in the respective response/vp_token examples. The examples are non-normative, of course, but they should be reasonably correct. Especially because much of that isn't explicitly defined in text so one has to infer from examples and text elsewhere. 

Related, shouldn't a credential profile explicitly define credential profile specific things? For example, I would have expected to see something saying the nonce and aud in the KB-JWT have to be the nonce and client_id values from the authz request.

Reply:
...and while discussing this with @bc-pi I noticed that `client_id_scheme` needs to be in the KB-JWT as well and actually in a lot of other places: https://github.com/openid/OpenID4VP/issues/124

Reply:
I have updated nonce and aud values of the kb-jwt to those of the request. The payload looks now as follows

```
{
  "nonce": "n-0S6_WzA2Mj",
  "aud": "https://client.example.org/cb",
  "iat": 1709839446,
  "sd_hash": "d-m8CkhGgx0KgSgNJvK5mR58GTR2hjhaHbBifUaPhKc"
}
```

I've also added a note in the spec to further clarify values of `aud` and `nonce`.

Reply:
> I am wondering if it is worth adding some clarification for sd_hash as well? 

Don't think that's needed here. sd_hash is fully specified in SD-JWT itself whereas the values for aud and nonce need to be specified by the protocol using  SD-JWT.